diff --git a/src/gl/texture_params.c b/src/gl/texture_params.c
index 4fbac86..63b0ae1 100644
--- a/src/gl/texture_params.c
+++ b/src/gl/texture_params.c
@@ -921,7 +921,7 @@ void realize_textures(int drawing)
             }
         }
         // check, if drawing, if mipmap needs some special care...
-        if(drawing) {
+        if(drawing && tex->glname) {
             if((globals4es.automipmap==3) || ((globals4es.automipmap==1) && (tex->mipmap_auto==0)) || (tex->compressed && (tex->mipmap_auto==0)))
                 tex->mipmap_need = 0;
             else
@@ -938,7 +938,8 @@ void realize_textures(int drawing)
                 tex->mipmap_done = 1;
             }
         }
-        realize_1texture(target, i, tex, glstate->samplers.sampler[i]);
+        if(tex->glname)
+            realize_1texture(target, i, tex, glstate->samplers.sampler[i]);
     }
     glstate->bound_changed = 0;
 }
